includes/blocks/class-sensei-blocks.php

```diff
@@ -20,6 +20,13 @@ class Sensei_Blocks {
 	 */
 	public $course;
 
+	/**
+	 * Lesson blocks.
+	 *
+	 * @var Sensei_Lesson_Blocks
+	 */
+	public $lesson;
```

Comment:
I know that we already do it for course but can we avoid exposing lesson as a public attribute? If we need to use it in another class we should supply it in the constructor. We should probably not even define a private attribute for it until we need it.

Reply:
This is kind of the approach we went with to keep blocks available to eachother. It's also nice to expose as much as possible for potential third-party extendability.

Reply:
Yep this is the existing approach but I think that there are many issues with it.

First we expose the data of the class as a public API which is against the open/closed principle. Another issue is that once you make an attribute public, you are pretty much stuck with it forever unless you use magic methods. Finally allowing a field to be accessed from anywhere, we introduce dependencies between components which are hard to track. Ideally any dependency of a class should be introduced in its constructor.

Extendability should be offered with the usage of interfaces or by the WP way of using filters. I don't think that we should allow this issue to creep into new code. We could potentially look into refactoring existing code to use private fields but I am not sure if it's worth it.

Reply:
Hey folks! I choose the cleaner solution for now, and I removed the public attribute of the lesson blocks: https://github.com/Automattic/sensei/pull/3886/commits/5bd620284c0a7a71da59864158ef0406f4ff2e1e

So if we need that in the future, we can check the best approach for that.

Reply:
Also, if we need that in the future, maybe we could transform the block classes in singleton :). Probably we won't want more than one instance of them.